nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1859,31 +1870,31 @@ define(
                         // this will not change, so we can just render it here.
                         PR.prettyPrint(null, container);
 
+                        const currentFsmState = fsm.getCurrentState().state;
+
                         // if we start out in 'new' state, we need to promote it to
                         // one of the 'editing' states
-                        if (fsm.getCurrentState().state.mode === 'new') {
+                        if (currentFsmState.mode === 'new') {
                             evaluateAppState();
                             return;
                         }
 
+                        let jobState = model.getItem('exec.jobState');
+                        // this should never happen but it may be present in old cells
+                        if (!jobState && model.getItem('exec.launchState.job_id')) {
+                            jobState = {
+                                job_id: model.getItem('exec.launchState.job_id'),
+                                status: 'created',
+                            };
+                        }
                         /*
-                         * Check the job state and request an update if it looks outdated.
+                         * Check the job state and request an update if it isn't a terminal status
                          */
-                        const jobState = model.getItem('exec.jobState');
-                        if (jobState && !Jobs.isValidJobStateObject(jobState)) {
+                        if (jobState && !Jobs.isTerminalStatus(jobState.status)) {
```

Comment:
replace all that FSM-based gobbledegook with using the job status to determine whether or not the cell needs to request job updates

Reply:
This can be replaced by the job manager (coming soon to a PR near you!)